flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-postgres/src/test/java/org/apache/flink/cdc/connectors/postgres/source/PostgresPipelineITCaseTest.java

```diff
@@ -135,6 +135,127 @@ public void testInitialStartupMode() throws Exception {
+                                        BinaryStringData.fromString("scooter"),
+                                        BinaryStringData.fromString("New scooter"),
+                                        5.5
+                                    })));
+
+            statement.execute(
+                    String.format(
+                            "INSERT INTO inventory.products (name, description, weight) VALUES ('football', 'New football', 6.6);",
+                            inventoryDatabase.getDatabaseName()));
+            expectedBinlog.add(
+                    DataChangeEvent.insertEvent(
+                            tableId,
+                            generator.generate(
+                                    new Object[] {
+                                        111, // next id after initialization
+                                        BinaryStringData.fromString("football"),
+                                        BinaryStringData.fromString("New football"),
+                                        6.6
+                                    })));
+
+            // Update existing data
+            statement.execute(
+                    String.format(
+                            "UPDATE inventory.products SET description = 'Updated description' WHERE id = 101;"));
+            expectedBinlog.add(
+                    DataChangeEvent.updateEvent(
+                            tableId,
+                            generator.generate(
+                                    new Object[] {
+                                        101,
+                                        BinaryStringData.fromString("scooter"),
+                                        BinaryStringData.fromString("Small 2-wheel scooter"),
+                                        3.14
+                                    }),
+                            generator.generate(
+                                    new Object[] {
+                                        101,
+                                        BinaryStringData.fromString("scooter"),
+                                        BinaryStringData.fromString("Updated description"),
+                                        3.14
+                                    })));
+
+            // Wait for the events to be processed
+            Thread.sleep(5_000);
+        }
+
+        // Collect the actual events
+        List<Event> actualEvents =
+                fetchResultsExcept(events, expectedBinlog.size(), createTableEvent);
+
+        // Filter out schema change events and keep only data change events
+        List<Event> actualDataChangeEvents =
+                actualEvents.stream()
+                        .filter(event -> event instanceof DataChangeEvent)
+                        .collect(Collectors.toList());
+
+        // Verify that we captured the expected number of data change events
+        assertThat(actualDataChangeEvents.size()).isGreaterThanOrEqualTo(expectedBinlog.size());
```

Comment:
It's better to add a scenario for creating a savepoint and restoring reads from the savepoint here too.